Release v0.1.0

ferbetanzo opened the pull request:
# Description

Preparing for releasing v.0.1.0 to pypi. It:

- Adds scripts to `/docs`.
- Updates pytket version to 1.34.0.
- Removes version constraints to black and pylint.
- Sets version to 0.1.0 in \_metadata.py
- Adds entry to changelog.

# Related issues

Please mention any github issues addressed by this PR.

# Checklist

- [X] I have performed a self-review of my code.
- [X] I have commented hard-to-understand parts of my code.
- [X] I have made corresponding changes to the public API documentation.
- [X] I have added tests that prove my fix is effective or that my feature works.
- [X] I have updated the changelog with any user-facing changes.

ferbetanzo committed:
Release v0.1.0

ferbetanzo committed:
fix: typo for pytket version in setup.py

ferbetanzo committed:
ci: use macos-latest instead of macos-12

CalMacCQ commented on docs/changelog.rst:
```diff
@@ -1,2 +1,8 @@
 ~~~~~~~~~
+
+0.1.0 (October 2024)
+-------------------
```
```suggestion
--------------------
```

This line is to short and sphinx complains

CalMacCQ commented on docs/build-docs.sh:
```diff
@@ -0,0 +1,21 @@
+EXTENSION_NAME="$(basename "$(dirname $(pwd))")"
+
+# Correct github link in navbar
+sed -i '' 's#CQCL/tket#CQCL/'$EXTENSION_NAME'#' _static/nav-config.js
```
This line is actually not needed as we no longer have the github link in the navbar

CalMacCQ commented on docs/build-docs.sh:
```diff
@@ -0,0 +1,21 @@
+sed -i '' 's#CQCL/tket#CQCL/'$EXTENSION_NAME'#' _static/nav-config.js
+
+# Build the docs. Ensure we have the correct project title.
+sphinx-build -b html -D html_title="$EXTENSION_NAME" . build
```
It looks like the docs build is failing locally due to a missing `index.rst` file.

If you rename the `intro.txt` file to `index.rst` this should fix it. You get the syntax highlighting too as well :)

Also for consistency with the other extensions we can add the pypi and github issue links to the bottom of `index.rst` as well

FOR PYTKET-QIR WE HAVE
```
.. toctree::
   :caption: Useful links

   Issue tracker <https://github.com/CQCL/pytket-qir/issues>
   PyPi <https://pypi.org/project/pytket-qir/>
```

CalMacCQ commented on docs/changelog.rst:
```diff
@@ -1,2 +1,8 @@
+0.1.0 (October 2024)
+-------------------
+
+* ``QuESTBackend`` that uses ``pyQuEST`` for running simulations in QuEST.
```
Totally optional suggestion below


You should be able to put a 

```
.. currentmodule:: pytket.extensions.quest
```

Directive at tthe top of the page

If you then do `:py:class:QuESTBackend` the crosslink to the docs should be generated.

CalMacCQ commented on docs/build-docs.sh:
```diff
@@ -0,0 +1,21 @@
+rm -rf build/
+
+# Move theming elements into the docs folder
+cp -R pytket-docs-theming/_static .
```
For the other extensions I've added a `docs/README.md` file linking to the pytket-docs-theming repo instructions

https://github.com/CQCL/pytket-qir/blob/main/docs/README.md

Totally optional

CalMacCQ reviewed (changes requested):
Looks good :) , just a few fixes to the docs build

CalMacCQ commented on .github/workflows/build_and_test.yml:
```diff
@@ -7,77 +7,77 @@ on:
+          name: artefact-${{ matrix.os }}
+          path: wheelhouse/
+      - name: Install docs dependencies
+        if: (matrix.os == 'ubuntu-22.04') && (github.event_name == 'pull_request' || github.event_name == 'schedule')
```
Maybe you want to look at how the docs build works in other extensions?

ferbetanzo committed:
chore: fix docs building

- Add currentmodule to changelog so links to docs work.
- Move intro.txt to index.rst.
- Add docs/README.md
- Fix doc building steps in CI.

ferbetanzo committed:
chore: fix ubuntu-22.04 instead of ubuntu-latest in build-and-test workflow

ferbetanzo committed:
chore: include submodules when checking out repo in CI

CalMacCQ reviewed (approved):
Docs build works great locally after the changes. Thanks Fernando. 

Maybe @cqc-melf  should double check that the other changes are consistent with the extensions before merging.

cqc-melf commented on .github/workflows/build_and_test.yml:
```diff
@@ -7,77 +7,82 @@ on:
     strategy:
       matrix:
-        os: ['ubuntu-22.04', 'macos-12', 'windows-2022']
+        os: ["ubuntu-22.04", "macos-latest", "windows-2022"]
```
```suggestion
        os: ["ubuntu-latest", "macos-latest", "windows-latest"]

cqc-melf commented on .github/workflows/build_and_test.yml:
```diff
@@ -90,72 +95,55 @@ jobs:
     name: Build docs
     if: github.event_name == 'release'
     needs: publish_to_pypi
     runs-on: ubuntu-22.04
```
```suggestion
    runs-on: ubuntu-latest

cqc-melf commented on .github/workflows/build_and_test.yml:
```diff
@@ -90,72 +95,55 @@ jobs:
     needs: distributed_tests
     runs-on: ubuntu-22.04
```
```suggestion
    runs-on: ubuntu-latest

cqc-melf reviewed (changes requested):
Looks good, might be good to update the other os versions, too. I have added suggestions.

cqc-melf commented on .github/workflows/lint.yml:
```diff
@@ -7,22 +7,21 @@ on:
+        run: pip install black pylint
+      - name: Check files are formatted with black
+        run: |
+          black --check .
```
It would be great if you could add a ruff check here, see for example https://github.com/CQCL/pytket-quantinuum/pull/497

ferbetanzo replied:
Done

ferbetanzo committed:
chore: Use latest OS versions in CI

Co-authored-by: cqc-melf <70640934+cqc-melf@users.noreply.github.com>

ferbetanzo committed:
chore: use ubuntu-latest

Co-authored-by: cqc-melf <70640934+cqc-melf@users.noreply.github.com>

ferbetanzo committed:
chore: use ubuntu-latest

Co-authored-by: cqc-melf <70640934+cqc-melf@users.noreply.github.com>

CalMacCQ commented on .github/workflows/build_and_test.yml:
```diff
@@ -88,74 +93,57 @@ jobs:
+      - name: Install poetry
+        run: pip install poetry
+      - name: Install extension
+        run: for w in `find wheelhouse/ -type f -name "*.whl"` ; do poetry install $w ; done
```
Ohh actually I think you might need to do `poetry run pip install` for the wheel rather than `poetry instal` here

ferbetanzo committed:
chore: acknowledge external contributors in README

ferbetanzo committed:
chore: install wheels with pip instead of poetry

ferbetanzo committed:
chore: add ruff check

ferbetanzo committed:
chore: fix ruff checks

ferbetanzo committed:
Merge branch 'chore/prepare-for-release' of github.com:CQCL/pytket-quest into chore/prepare-for-release

ferbetanzo committed:
chore: fix black formatting

ferbetanzo committed:
chore: fix ubuntu version checks

CalMacCQ commented on .github/workflows/docs/conf.py:
```diff
@@ -44,7 +42,7 @@
 # The following code is for resolving broken hyperlinks in the doc.
```
This file can actually be deleted along with the entire `.github/workflows/docs` directory. 

I can do this in a follow up PR if preferred.

ferbetanzo replied:
Done

ferbetanzo committed:
chore: remove .github/workflows/docs/ dir

CalMacCQ reviewed (approved):
Looks great to me :) thanks again
